core/src/main/java/com/datastax/oss/driver/internal/core/config/composite/CompositeDriverConfigLoader.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.config.composite;
+
+import com.datastax.oss.driver.api.core.config.DriverConfig;
+import com.datastax.oss.driver.api.core.config.DriverConfigLoader;
+import com.datastax.oss.driver.api.core.context.DriverContext;
+import com.datastax.oss.driver.internal.core.util.concurrent.CompletableFutures;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+import java.util.concurrent.CompletionStage;
+
+public class CompositeDriverConfigLoader implements DriverConfigLoader {
+
+  private final DriverConfigLoader primaryConfigLoader;
+  private final DriverConfigLoader fallbackConfigLoader;
+
+  public CompositeDriverConfigLoader(
+      @NonNull DriverConfigLoader primaryConfigLoader,
+      @NonNull DriverConfigLoader fallbackConfigLoader) {
+    this.primaryConfigLoader = Objects.requireNonNull(primaryConfigLoader);
+    this.fallbackConfigLoader = Objects.requireNonNull(fallbackConfigLoader);
+  }
+
+  @NonNull
+  @Override
+  public DriverConfig getInitialConfig() {
+    DriverConfig primaryConfig = primaryConfigLoader.getInitialConfig();
+    DriverConfig fallbackConfig = fallbackConfigLoader.getInitialConfig();
+    return new CompositeDriverConfig(primaryConfig, fallbackConfig);
+  }
+
+  @Override
+  public void onDriverInit(@NonNull DriverContext context) {
+    fallbackConfigLoader.onDriverInit(context);
+    primaryConfigLoader.onDriverInit(context);
+  }
+
+  @NonNull
+  @Override
+  public CompletionStage<Boolean> reload() {
+    if (!primaryConfigLoader.supportsReloading() && !fallbackConfigLoader.supportsReloading()) {
+      return CompletableFutures.failedFuture(
+          new UnsupportedOperationException(
+              "Reloading is not supported (this is a composite config, "
+                  + "and neither the primary nor the fallback are reloadable)"));
+    } else if (!primaryConfigLoader.supportsReloading()) {
+      return fallbackConfigLoader.reload();
+    } else if (!fallbackConfigLoader.supportsReloading()) {
+      return primaryConfigLoader.reload();
+    } else {
+      return fallbackConfigLoader
```

Comment:
- is there a specific reason for this order of reloading? ( I mean that fallback first, then primary)
- to double-check: the `thenCompose` will execute also if the fallback `reload()` will fail, right?
- Should we log it if there will be an error while reloading? (for example this:
https://github.com/datastax/java-driver/pull/1445/files#diff-dc57c9b73b57019d14f9babae5c5d7b2R72-R77)

Reply:
The order does not matter, but you bring a good point about error handling, `thenCompose` does not execute the callback if the first future is failed. I'll fix the code to reload both independently and only combine the result after.

If an error occurs, it will be passed to the caller via the resulting stage, so we should not log it.